src/cdk-experimental/scrolling/virtual-scroll-viewport.ts

```diff
@@ -319,4 +286,40 @@ export class CdkVirtualScrollViewport implements DoCheck, OnInit, OnDestroy {
     this._viewportSize = this.orientation === 'horizontal' ?
         viewportEl.clientWidth : viewportEl.clientHeight;
   }
+
+  /** Queue up change detection to run. */
+  private _markForCheck(runAfter?: Function) {
+    if (runAfter) {
+      this._runAfterCheck.push(runAfter);
+    }
+    if (!this._checkPending) {
+      this._checkPending = true;
+      this._ngZone.runOutsideAngular(() => Promise.resolve().then(() => {
+        if (this._ngZone.isStable) {
+           this._doCheck();
+        } else {
+          this._ngZone.onStable.pipe(take(1)).subscribe(() => this._doCheck());
+        }
+      }));
+    }
+  }
+
+  /** Run change detection. */
+  private _doCheck() {
+    this._checkPending = false;
+    this._ngZone.run(() => this._changeDetectorRef.detectChanges());
+    // In order to batch setting the scroll offset together with other DOM writes, we wait until a
```

Comment:
I don't quite follow what this comment means about waiting, since `detectChanges` is synchronous

Reply:
Yes, but in the `_markForCheck` method above I'm batching them up using a `Promise.resolve()`. This way if we change multiple properties it waits until after they're all updated and calls `detectChanges` once

Reply:
Should the comment live where the wait is scheduled?

Reply:
I split the comment up to explain the batching part where it actually happens